### DISCUSSION TITLE
Add instructions how to build from source with more distributions

### DISCUSSION THREAD

**Discussion by wehagy** [Vysp3r/ProtonPlus#51] [Ideas]
I'm thinking of adding how to build from source with more distributions (ubuntu, debian "testing or sid, stable not works" and maybe arch), but I'm not sure that's a good thing as it could end up being a burden, so I think it's relevant to discuss and know what you think @Vysp3r.

**Comment by Vysp3r** [comment]
I mean I've added a link to each dependencies so that people using something else then Fedora can easily find them for their distribution. I guess we could add the most used distributions like Ubuntu and Arch. I think adding more would just be too much. Maybe once I make the Wiki we could move that section in the Wiki and add way more distribution, but for now I think just those two would be good enough.

**Reply by wehagy** [reply]
Cool, I'm going to do the instructions on how to compile from source with Arch and with Ubuntu, I'll soon open two issues to track this (need to finish the issue #49 first), but to be honest I wanted to do "debian and derivatives" and "arch and derivatives", that would cover the the vast majority of distros, but almost certainly debian bullseye does not work, very ancient and the necessary packages do not exist in the official repository nor in the backports, and I will not recommend anyone to use testing/sid without knowing what they're doing, much less a FrankenDebian.

**Reply by Vysp3r** [reply]
If you think Debian would be better I'm all for it.

**Reply by wehagy** [reply]
I made some quick tests.

- The debian bullseye don't work, a lot of packages missing.
- Ubuntu 22.04 not working due to this version dependency.
```
ERROR: Invalid version of dependency, need 'libadwaita-1' ['>= 1.2'] found '1.1.0'.
```
- Ubuntu 22.10 compiles and with some quick testing is working, but I have that warning on compiling, maybe it's harmless.
```
Program gtk4-update-icon-cache found: NO
Program gtk-update-icon-cache found: YES (/usr/bin/gtk-update-icon-cache)
```

\
I think the debian bullseye support and ubuntu 22.04 is a won't fix problem, but ubuntu 22.10 and derivatives onwards is fine.

**Reply by Vysp3r** [reply]
Indeed, if they don't have the minimum requirements for ProtonPlus dependencies, we can't do much about that. I will not remove features so that older versions of distro's supports it.

**Reply by wehagy** [reply]
As soon as I finish #49 I'll start working on it, probably through the arch that has newer packages.
